tools/shield_utils.py

```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script is intended to be an entry point for `devtool test`.
+# It provides the needed primitives for creation of a "shielded"
+# environment setup. Shielding is based entirely on cgroup cpuset.
+#
+# The idea behind the environment setup is to isolate and shield workloads
+# started by the testing framework. To this end, we will be interested in
+# three high-level cpusets:
+# * root - the root cpuset hierarchy.
+# * system - where existing tasks will run.
+# * docker - where tests workloads will run.
+#
+# Docker shares all the available cpus and memory nodes with root, but
+# when containers start, they can receive a slice of the available cpus
+# and memory, depending on what is requested by the "caller". The slice
+# must be contained by an isolated NUMA node. Shielding this slice of the
+# NUMA node is done by moving all the existing tasks inside system cpuset,
+# letting the "shielded" and free for available for a new docker container.
+
+
+import os
+import re
+import io
+import sys
+import random
+import string
+# Add the file directory to PATH to be able to import other modules from within
+# the same directory.
+sys.path.append(os.path.dirname(__file__))
+import log_utils as log
+
+
+# Exit codes.
+EINVPATTERN = 1
+EINVSYSFS = 2
+ENOAVAILCPU = 3
+ECONFLICTCPUSET = 4
+EINVARG = 5
+ENOAVAILMEM = 6
+EOFFNODE = 7
+EDIRTYENV = 8
+ESHARENODE = 9
+ETEARDOWNSYS = 10
+EEXCEPT = 11
+ESKIP = 12
+
+
+# Patterns.
+CPUSET_MOUNT_PATTERN="^[^ ]+ (/.+) (?:cpuset |cgroup (?:[^ ]*,)?cpuset[, ])"
+SYSFS_MOUNT_PATTERN="sysfs (.+) sysfs .+"
+RANGE_PATTERN="([0-9][1-9]*)-([0-9][1-9]*)"
+
+
+def _locate_mount_point(pattern):
+    mount_point_regex = re.compile(pattern)
+    path = None
+    with open("/proc/mounts") as fp:
+        for line in fp:
+            res = mount_point_regex.search(line)
+            if res:
+                path = res.group(1)
+                break
+
+    return path
+
+
+# Parse list formats for cpuset and mems.
```

Comment:
Instead of adding function descriptions as comments, you should enclose the comments in pydoc comments:

```
This is a python block
```